openstf/model/ml_model/abstract_ml_model.py

```diff
@@ -0,0 +1,84 @@
+# SPDX-FileCopyrightText: 2017-2021 Alliander N.V. <korte.termijn.prognoses@alliander.com> # noqa E501>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+from abc import ABC, abstractmethod
+import pandas as pd
+import structlog
+from typing import Union
+import numpy as np
+from sklearn.base import RegressorMixin, BaseEstimator
+
+
+class AbstractMLModel(BaseEstimator, RegressorMixin, ABC):
+    """
+    This class defines an abstract ML model,
+    of which concrete implementation can be used as a workhorse throughout openstf
+
+    If you want to include a new ML model in openstf, it should adhere to the interface
+    defined here
+    """
+
+    def __init__(self) -> None:
+        self.logger = structlog.get_logger(self.__class__.__name__)
+
+    @abstractmethod
+    def predict(self, x: pd.DataFrame) -> np.array:
+        pass
+
+    @abstractmethod
+    def fit(self, x: np.array, y: np.array, **kwargs) -> RegressorMixin:
+        pass
+
+    def score(self, forecast: np.array, actual: np.array) -> float:
+        """By default, score returns the R^2 score."""
+        return RegressorMixin.score(forecast, actual)
+
+    @abstractmethod
+    def set_params(self, **kwargs) -> None:
+        """Method to set hyperparams of the model, which are used in 'fit'"""
+        pass
+
+    ##### Currently, feature importance is calculated quite xgb/lgb specific.
+    # It requires that:
```

Comment:
Twee functionele wensen:
- vastleggen welke features gebruikt zijn bij trainen; dit wordt gebruikt bij het maken van een nieuwe prognose
- overzicht van feature_importance. Eenerzijds voor gebruikers om inzicht te geven in het model, anderzijds om de gewogen compleetheid van de input te bepalen.